### PR TITLE
Avoid exposing `chainerx.Scalar` to Python

### PR DESCRIPTION
**Pull request by asi1024**
~Depends on #6481.~

### CODE REVIEW

**Comment by asi1024** [comment]
Rebased.

File: chainerx_cc/chainerx/python/scalar.cc [hunk 1]
```diff
@@ -18,34 +18,10 @@ namespace python_internal {
 namespace py = pybind11;  // standard convention
 
 void InitChainerxScalar(pybind11::module& m) {
-    py::class_<Scalar> c{m, "Scalar"};
+    py::class_<Scalar> c{m, "_Scalar"};
```
Comment (niboshi) [inline chainerx_cc/chainerx/python/scalar.cc]: Is this binding still needed?

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**